File: training-data/python/matrix_multiply.py
from typing import List


def multiply(a: List[List[float]], b: List[List[float]]) -> List[List[float]]:
    rows_a, cols_a = len(a), len(a[0])
    rows_b, cols_b = len(b), len(b[0])
    if cols_a != rows_b:
        raise ValueError("incompatible matrix dimensions")

    result = [[0.0] * cols_b for _ in range(rows_a)]
    for i in range(rows_a):
        for j in range(cols_b):
            result[i][j] = sum(a[i][k] * b[k][j] for k in range(cols_a))
    return result


if __name__ == "__main__":
    a = [[1, 2], [3, 4]]
    b = [[5, 6], [7, 8]]
    print(multiply(a, b))

File: training-data/go/matrix_multiply.go
package main

import "fmt"

func multiply(a, b [][]int) [][]int {
	rowsA, colsA, colsB := len(a), len(a[0]), len(b[0])
	result := make([][]int, rowsA)
	for i := range result {
		result[i] = make([]int, colsB)
		for j := 0; j < colsB; j++ {
			sum := 0
			for k := 0; k < colsA; k++ {
				sum += a[i][k] * b[k][j]
			}
			result[i][j] = sum
		}
	}
	return result
}

func main() {
	a := [][]int{{1, 2}, {3, 4}}
	b := [][]int{{5, 6}, {7, 8}}
	fmt.Println(multiply(a, b))
}
